projects/fhi-angular-highcharts/src/lib/fhi-angular-highcharts.component.ts

```diff
@@ -180,6 +186,23 @@ export class FhiAngularHighchartsComponent implements OnChanges {
     this.diagramTypeGroups_NEW = this.diagramTypeGroupService.getDiagramTypeGroups();
   }
 
+  private updateDecimals() {
+    const unit = this.allDiagramOptions.unit;
+    let decimals = unit?.length >= 0 ? unit[0].decimals : undefined; // Currently only support for one unit
+
+    // Temporary fallback before this.allDiagramOptions.decimals deprecates in v5
+    if (decimals === undefined) {
+      decimals = this.allDiagramOptions.decimals;
+    }
```

Comment:
this could be one line


`let decimals = unit?.length >= 0 ? unit[0].decimals : this.allDiagramOptions.decimals;`